packages/dev/src/main.ts

```diff
@@ -236,7 +255,13 @@ export class NetlifyDev {
     return config
   }
 
-  async handle(request: Request) {
+  async handle(request: Request, options: HandleOptions = {}) {
+    const result = await this.handleAndIntrospect(request, options)
+
+    return result?.response
+  }
+
+  async handleAndIntrospect(request: Request, options: HandleOptions = {}) {
```

Comment:
A new entry point that is the same as `handle` but also provides additional information, which for now is just the type of match being returned.